cloud/amazon/ecs_taskdefinition.py

```diff
@@ -120,14 +122,35 @@ def __init__(self, module):
                 module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
             self.ecs = boto3_conn(module, conn_type='client', resource='ecs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
         except boto.exception.NoAuthHandlerFound, e:
-            module.fail_json(msg="Can't authorize connection - "+str(e))
+            self.module.fail_json(msg="Can't authorize connection - "+str(e))
 
     def describe_task(self, task_name):
         try:
             response = self.ecs.describe_task_definition(taskDefinition=task_name)
             return response['taskDefinition']
-        except botocore.exceptions.ClientError:
-            return None
+        except botocore.exceptions.ClientError, e:
+            self.module.fail_json(msg="Can't describe task - "+str(e))
```

Comment:
This exception is raised when trying to describe a task that doesn't exist. Failing here means we can't create new task definitions with this module.

I think something like this should work:

```python
def describe_task(self, task_name):
    try:
        families_response = self.ecs.list_task_definition_families()
        if task_name not in families_response['families']:
            return None

        response = self.ecs.describe_task_definition(taskDefinition=task_name)
        return response['taskDefinition']
    except botocore.exceptions.ClientError, e:
        self.module.fail_json(msg="Can't describe task - "+str(e))
```